lib/strategy/whitelist.rb

```diff
@@ -19,9 +19,17 @@ def process_record(index, record)
         @primary_keys.each do |key|
           dest_record[key] = record[key]
         end
-        dest_record.save!
+        if bulk_process?
+          collect_for_bulk_process(dest_record)
+        else
+          dest_record.save!
+        end
       end
 
+      def bulk_store(records)
+        columns = source_table.column_names
+        source_table.import @primary_keys + columns, records, validate: false, on_duplicate_key_update: columns
```

Comment:
I know this is an old PR, but was useful for me, thanks!

I think this should be `dest_table` not `source_table`, though, and `@primary_keys` isn't needed. I also want to perseveres timestamps So:
`dest_table.import columns, records, validate: false, on_duplicate_key_update: columns, timestamps: false`

Reply:
Yes, you are right. For my usecases source and dest were the same, so I didn't recognized this problem. Thanks.
I updated my PR.

Reply:
The `@primary_keys` is needed, if you want to update the records in place, I think. At least for me it fails, if I don't add them explicitly, like I have done.